tests/backup/backup_test.go

```diff
@@ -1937,35 +1968,62 @@ func getProviderClusterConfigPath(provider string, kubeconfigs []string) (string
 
 // CreateBackup creates backup
 func CreateBackup(backupName string, clusterName string, bLocation string, bLocationUID string,
-	namespaces []string, labelSelectors map[string]string, orgID string) {
-
-	Step(fmt.Sprintf("Create backup [%s] in org [%s] from cluster [%s]",
-		backupName, orgID, clusterName), func() {
+	namespaces []string, labelSelectors map[string]string, orgID string, uid string, pre_rule_name string,
+	pre_rule_uid string, post_rule_name string, post_rule_uid string) {
 
-		backupDriver := Inst().Backup
-		bkpCreateRequest := &api.BackupCreateRequest{
-			CreateMetadata: &api.CreateMetadata{
-				Name:  backupName,
-				OrgId: orgID,
-			},
-			BackupLocationRef: &api.ObjectRef{
-				Name: bLocation,
-				Uid:  bLocationUID,
-			},
-			Cluster:        clusterName,
-			Namespaces:     namespaces,
-			LabelSelectors: labelSelectors,
+	var bkp *api.BackupObject
+	var bkp_uid string
+	backupDriver := Inst().Backup
+	bkpCreateRequest := &api.BackupCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  backupName,
+			OrgId: orgID,
+		},
+		BackupLocationRef: &api.ObjectRef{
+			Name: bLocation,
+			Uid:  bLocationUID,
+		},
+		Cluster:        clusterName,
+		Namespaces:     namespaces,
+		LabelSelectors: labelSelectors,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  uid,
+		},
+		PreExecRuleRef: &api.ObjectRef{
+			Name: pre_rule_name,
+			Uid:  pre_rule_uid,
+		},
+		PostExecRuleRef: &api.ObjectRef{
+			Name: post_rule_name,
+			Uid:  post_rule_uid,
+		},
+	}
+	//ctx, err := backup.GetPxCentralAdminCtx()
+	ctx, err := backup.GetAdminCtxFromSecret()
+	dash.VerifyFatal(err, nil, "Fetching px-central-admin ctx")
```

Comment:
if this need to be posted to dashboard, plz use log.FailOnError

Reply:
done